stm32_base/hal/imu_stm32/invensense_9150.c

```diff
@@ -2,46 +2,214 @@
 #include <invensense_9150.h>
 #include <stdint.h>
 #include <stddef.h>
-
+#include "taskUtil.h"
+#include "printk.h"
+#include <math.h>
 static struct  {
     struct i2c_dev *i2c;
     uint8_t addr;
+    uint8_t mag_addr;
 } is9150_dev;
 
+static float mag_sens_adj[3];
+
 static int is9150_readreg(uint8_t reg_addr, uint8_t *reg_val)
 {
-    int res = 0;
+    int res;
     res = i2c_read_reg8(is9150_dev.i2c, is9150_dev.addr,
                         reg_addr, reg_val);
     return res;
 }
 
+static int is9150_write_reg(uint8_t reg_addr, uint8_t reg_val)
+{
+    int res;
```

Comment:
Nit feedback.  Do
```C
const int res = i2c_write_reg8(is9150_dev.i2c, is9150_dev.addr, reg_addr, reg_val);
```
instead as the compiler can optimize that better and you can still use it in debug statements as needed.